### ISSUE TITLE
Refactor:: Move to a new File incorrectly adds export keyword to source file

### ISSUE THREAD

**Issue by hammus** [Microsoft/vscode#51815]
Hi folks, there doesn't seem to be an open issue for this.

When using Refactor -> Move to a new File on a TypeScript class that extends an `abstract` Base class, I'm getting some really weird `export` additions added to **non-**abstract members on the Base class.

- VSCode Version: 
```
Version 1.24.1
Commit 24f62626b222e9a8313213fb64b10d741a326288
Date 2018-06-13T17:51:32.889Z
Shell 1.7.12
Renderer 58.0.3029.110
Node 7.9.0
Architecture x64
```
- OS Version: 
**Windows 10 Pro x64 Build 17134**

Steps to Reproduce:

1. Here is a minimal reproduction of my abstract class
```typescript
export abstract class FeatureDetector {
    public abstract readonly Independents: string[];
    public abstract readonly Dependents: string[];

    public IsIndependent(features: IFeatureSubset): boolean {
         // method definition
    }

    public HasDependents(features: IFeatureSubset): boolean {
        return !this.IsIndependent(features);
    }
}
```
2. I Used Refactor -> Move to a new file on a derived class `FAODetector`:
```typescript
export class FAODetector extends FeatureDetector {
 // class definition
}
```

3. Once the Refactor is complete the **base** class has been modified in some unusual ways:
```typescript
export abstract class FeatureDetector {
   // Member definitions

    public HasDependents(features: IFeatureSubset): boolean {
        return !this.export IsIndependent(features); // <-- Note the export added to this.IsIndependent
    }
}
```

4. There are actually a bunch of other strange exports added that I need to remove too but I havent added these examples as I suspect they may be result of the idiosyncrasies of my code and that the code is for an Electron app with Typescript, and TS can sometimes be a little finicky with the mixed Node / Browser environment. But for completeness here is another example:
```typescript
export {fao as FAODetector };export export // <-- Bit too keen on the exports methinks.
```

Does this issue occur when all extensions are disabled?: **Yes**
Confirmed this issue is present with Code running with `--disable-extensions` flag.

**Comment by mjbvz** [comment]
Please test this in the latest VS Code insiders build as well. We just fixed a similar bug

**Comment by mjbvz** [comment]
Try 11b0f3a40be055e02b721abbb93cb0012818f99e. That build is two days old

**Comment by hammus** [comment]
Seems fixed for me in Insiders 11b0f3a40be055e02b721abbb93cb0012818f99e

**Comment by mjbvz** [comment]
Duplicate of https://github.com/Microsoft/TypeScript/issues/24643

**Comment by timfish** [comment]
I still get this or something similar with this insiders build:
```
Version: 1.25.0-insider
Commit: 6ca99aca80e80224f1c0bd4370c6ec4941c3bc54
Date: 2018-06-29T09:03:04.490Z
Electron: 1.7.12
Chrome: 58.0.3029.110
Node.js: 7.9.0
V8: 5.8.283.38
Architecture: x64
TypeScript: 2.9.2:
```

After in source:
![image](https://user-images.githubusercontent.com/1150298/42126547-e0fe19ca-7c8a-11e8-86cf-625347119d59.png)

In new Destination:
![image](https://user-images.githubusercontent.com/1150298/42126552-0355f2ae-7c8b-11e8-9182-d3b491ed026d.png)

**Comment by mjbvz** [comment]
@timfish You are correct. The fix was mistakenly not taken into 2.9.2.

It should be fixed in typescript@next however. You can upgrade your workspace to use that version by following [these instructions](https://code.visualstudio.com/Docs/languages/typescript#_using-newer-typescript-versions)
